portal/report/document_downloads_action.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+/**
+ * Download selected patient documents in a zip file
+ *
+ * @package   OpenEMR
+ * @link      https://www.open-emr.org
+ * @author    Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2024 Jerry Padgett <sjpadgett@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+require_once("../verify_session.php");
+require_once("$srcdir/documents.php");
+require_once($GLOBALS['fileroot'] . "/controllers/C_Document.class.php");
+
+use OpenEMR\Common\Csrf\CsrfUtils;
+
+if (!CsrfUtils::verifyCsrfToken($_POST['csrf_token_form'] ?? '')) {
+    CsrfUtils::csrfNotVerified();
+}
+
+// Check if documents are selected
+if (empty($_POST['documents'])) {
+    die("No documents selected.");
+}
+
+// Get the temporary folder
+$tmp = $GLOBALS['temporary_files_dir'];
+$documentIds = $_POST['documents'];
```

Comment:
Won't this allow a patient to grab documents of other patients?

Reply:
When collect documents below would ensure that the patient id of the document matches the below $pid

Reply:
And if not, then would die the script horribly since likely a bad actor